tests/PSRule.Rules.Azure.Tests/Azure.Deployment.Tests.ps1

```diff
@@ -177,4 +177,62 @@ Describe 'Azure.Deployment.OuterSecret' -Tag 'Deployment' {
 
 }
 
+Context 'Resource name - Azure.Deployment.Name' {
+    BeforeAll {
+        $invokeParams = @{
+            Baseline = 'Azure.All'
+            Module = 'PSRule.Rules.Azure'
+            WarningAction = 'Ignore'
+            ErrorAction = 'Stop'
+        }
+
+        $testObject = [PSCustomObject]@{
+            Name = ''
+            ResourceType = 'Microsoft.Resources/deployments'
+        }
+    }
+
+    BeforeDiscovery {
+        $validNames = @(
+            'nestedtemplate'
+            'nestedtemplate1'
+            'nestedtemplate-1'
+            'NESTEDTEMPLATE'
+            'nestedtemplate(1)'
+            'linkedtemplate'
+            'linkedtemplate1'
+            'linkedtemplate-1'
+            'LINKEDTEMPLATE'
+            'linkedtemplate(1)'
```

Comment:
Not sure if these add value since they are basically the same name tests as the `nestedtemplate` checks. We can a `.` check instead.

```suggestion
            'nested.Template1'
```